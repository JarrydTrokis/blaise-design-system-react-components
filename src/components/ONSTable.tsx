import React, { ReactElement } from "react";

interface ONSTable {
  columns: string[]
  children: ReactElement
  tableID?: string
  tableCaption?: string
}

export const ONSTable = ({ columns, children, tableCaption, tableID }: ONSTable): ReactElement => {
  return (
    <>
      <table
        className="table"
        data-testid={tableID}
        id={tableID}>
        {tableCaption && <caption className="table__caption">{tableCaption}</caption>}
        <thead className="table__head">
          <tr className="table__row">
            {
              columns.map((title: string, index: number) => (
                <th scope="col" className="table__header" key={`${title}-${index}`}>{title}</th>
              ))
            }
          </tr>
        </thead>
        <tbody className="table__body">
          {children}
        </tbody>
      </table>
    </>
  );
};


export default ONSTable;
