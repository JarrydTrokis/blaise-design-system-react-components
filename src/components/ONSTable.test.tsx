import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import ONSTable from "./ONSTable";

describe("ONSTable", () => {
    it("renders a formatted table", async () => {
        render(
            <ONSTable columns={["column1", "column2", "column3"]} tableID="my-test-table">
                <tr>
                    <td>Value column 1</td>
                    <td>Value column 2</td>
                    <td>Value column 3</td>
                </tr>
            </ONSTable>,
        );

        await waitFor(() => {
            expect(screen.getByTestId("my-test-table")).toMatchSnapshot();
        });
    });
});
