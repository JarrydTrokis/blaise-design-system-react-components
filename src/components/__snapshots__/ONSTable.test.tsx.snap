// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONSTable renders a formatted table 1`] = `
<table
  class="table"
  data-testid="my-test-table"
  id="my-test-table"
>
  <thead
    class="table__head"
  >
    <tr
      class="table__row"
    >
      <th
        class="table__header"
        scope="col"
      >
        column1
      </th>
      <th
        class="table__header"
        scope="col"
      >
        column2
      </th>
      <th
        class="table__header"
        scope="col"
      >
        column3
      </th>
    </tr>
  </thead>
  <tbody
    class="table__body"
  >
    <tr>
      <td>
        Value column 1
      </td>
      <td>
        Value column 2
      </td>
      <td>
        Value column 3
      </td>
    </tr>
  </tbody>
</table>
`;
